### PR TITLE
Add unit tests for refactored segmentation & prevent saving .dot in root

### PR DESCRIPTION
**Pull request by JihwanEom**
### Summary
- Add unit tests for refactored `OTXSegmentationTask`, `MMSegmentationTask`
- Remove unnecessary code lines in `configure_data`
- Bugfix in `_infer_model` related with `dump_soft_prediction` (to be identical with previous code)
- **Prevent saving `original_graph.dot` & `compressed_graph.dot` in root folder when run tests for all tasks**.
- **Prevent saving `dummy.converted.pth`in root folder when run detection unit test**

#### unit tests
image

#### integration test (segmentation)
image

#### E2E test (segmentation)
image

### How to test
```
pytest -s -v tests/unit/
pytest -s -v tests/integration/cli/segmentation
pytest -s -v tests/e2e/cli/segmentation
```

### Checklist

- [x] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added e2e tests for validation.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [x] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [x] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2023 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

File: otx/algorithms/classification/configs/efficientnet_b0_cls_incr/compression_config.json [hunk 1]
```diff
@@ -3,7 +3,7 @@
     "find_unused_parameters": true,
     "nncf_config": {
       "compression": [],
-      "log_dir": "."
+      "log_dir": "/tmp"
```
Comment (harimkang) [inline otx/algorithms/classification/configs/efficientnet_b0_cls_incr/compression_config.json:6]: Are there any other issues with modifying this to `tmp`?

**Review by harimkang — DISMISSED** [review]

File: otx/algorithms/classification/configs/efficientnet_b0_cls_incr/compression_config.json (hunk 1 above)
Comment (JihwanEom) [reply otx/algorithms/classification/configs/efficientnet_b0_cls_incr/compression_config.json:6]: For my best knowledge, no. Anomaly templates already have `log_dir` as `/tmp`.

**Review by supersoob — DISMISSED** [review]
Nice work. Thanks.

File: tests/unit/algorithms/segmentation/adapters/mmseg/test_mmseg_configurer.py
Comment (eunwoosh) [inline tests/unit/algorithms/segmentation/adapters/mmseg/test_mmseg_configurer.py:1]: `@e2e_pytest_unit` is missing at some of tests

File: tests/unit/algorithms/segmentation/adapters/test_otx_segmentation_task.py
Comment (eunwoosh) [inline tests/unit/algorithms/segmentation/adapters/test_otx_segmentation_task.py:1]: `@e2e_pytest_unit` is missing at some of tests

File: tests/unit/algorithms/segmentation/test_task.py
Comment (eunwoosh) [inline tests/unit/algorithms/segmentation/test_task.py:1]: `@e2e_pytest_unit` is missing at some of tests

**Review by eunwoosh — APPROVED** [review]

**Review by sungmanc — APPROVED** [review]

**Pull request merged**